lang/csharp/src/apache/main/CodeGen/CodeGenUtil.cs

```diff
@@ -132,5 +144,21 @@ public string UnMangle(string name)
                     builder.Append(name[i]);
             return builder.ToString();
         }
+
+        private CodeAttributeDeclaration GetGeneratedCodeAttribute()
```

Comment:
It uses `[GeneratedCodeAttribute("avrogen", "1.12.0.0")]`  Why exactly, would you want to use global in this case?

Reply:
1. Unfortunately `1.12.0-rc.1`, `1.12.0-rc.2` and `1.12.0` will be shown as `1.12.0.0`. Which seems to be incorrect behavior. Please change it to the full version,

2, Namespaces are getting tricky, especially with the latest global usings. It is preferred to use `::global` just to avoid any issue if someone  is implementing its own `GeneratedCodeAttribute`. E.g. Google protobuf is doing the same.

Reply:
1.  A little unsure on how to get the updated versioning, since this is how the project has been configured as well as the version is the same one we use in the ` <auto-generated>` header. `System.Reflection.Assembly.GetExecutingAssembly().GetName().Version`
2. Added `global::` Which is just odd to me since this used to be a big thing not to do with things under System.  I know there are changes to support how .Net 6 is doing things, which means a lot more global usings.

Reply:
Get the InformationalVersion attribute for the full version. IMO the FileVersion or AssembleVersion are pretty much useless, since it does not support SemVer. Since this project supports pre-release label, we should use it. I would even change it in the header comments, since that version does not reflect the real version (if pre-released). I understand that pre-release version is not pushed to nuget, however it is released in github,

IMO using the proper and full  version makes your life easier when you implement the unit tests. E.g. whatever avrogen --version returns, should be the version tag written in the cs file. Otherwise it is confusing for the end user if avrogen --version returns something different than the version tag written in the generated cs file

Reply:
[global::System.CodeDom.Compiler.GeneratedCodeAttribute("avrogen", "1.12.0-SNAPSHOT")]

Reply:
To make your life more complicated:

```
// VERSION.txt = 1.12.1-SNAPSHOT
string version = Assembly.GetExecutingAssembly().GetName().Version.ToString(); // => 1.12.0.0 INCORRECT
string fullVersion = Assembly.GetExecutingAssembly().GetCustomAttribute<AssemblyInformationalVersionAttribute>().InformationalVersion.ToString(); // => 1.12.1-SNAPSHOT CORRECT
```

FileVersion is set in common.props, which is defined as `<FileVersion>$(MajorVersion).$(MinorVersion).$(BuildNumber).0</FileVersion>` which IMO is incorrect and it should be `<FileVersion>$(MajorVersion).$(MinorVersion).$(PatchNumber).0</FileVersion>`.

Probably it is a good idea to fix in another ticket, because the e.g. 1.11.1, the version written into the cs files will be 1.11.0 :(. It definetely makes sense not to tryst thany other version, except he InformationalVersion

Reply:
Yeah, I dug through and found it.  The update makes sense and makes things more consistent.